eng/references.targets

```diff
@@ -1,67 +1,52 @@
-    <ItemGroup>
-      <DefaultReferenceExclusions Include="@(ReferenceFromRuntime)"/>
-
-      <!-- Reference everything in the targeting pack directory -->
-      <DefaultReferenceDirs Include="$(RefPath)" />
-      <DefaultReferenceItems Include="%(DefaultReferenceDirs.Identity)/*.dll" />
+  <!-- netstandard is a default reference whenever building an implementation assembly for .NETCoreApp. -->
+  <ItemGroup Condition="'$(IncludeDefaultReferences)' == 'true' and
+                        '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and
+                        '$(IsSourceProject)' == 'true'">
+    <Reference Include="netstandard"
+               Private="false" />
+  </ItemGroup>
 
-      <DefaultReferenceExclusions>
-        <RefDir>%(DefaultReferenceDirs.Identity)</RefDir>
-      </DefaultReferenceExclusions>
-      <_defaultReferenceExclusionsFullPath Include="%(DefaultReferenceExclusions.RefDir)%(DefaultReferenceExclusions.Identity).dll" />
+   <!-- If a CoreLib ProjectReference is present, make all P2P assets non transitive. -->
+  <ItemGroup Condition="'@(ProjectReference)' != '' and
+                         @(ProjectReference->AnyHaveMetadataValue('Identity', '$(CoreLibProject)'))">
+    <ProjectReference Update="@(ProjectReference)"
+                      PrivateAssets="all" />
+  </ItemGroup>
 
-      <!-- Ensure conflict resolution can see these references. -->
-      <Reference Include="%(DefaultReferenceItems.FullPath)" Private="false" />
+  <!-- Filter out transitive P2Ps which should be excluded. -->
+  <Target Name="FilterTransitiveProjectReferences"
+          AfterTargets="IncludeTransitiveProjectReferences"
+          Condition="'$(DisableTransitiveProjectReferences)' != 'true' and
+                     '@(DefaultReferenceExclusion)' != ''">
+    <ItemGroup>
+      <_transitiveProjectReferenceWithExclusion Include="@(ProjectReference)">
+        <Exclusion>%(DefaultReferenceExclusion.Identity)</Exclusion>
+      </_transitiveProjectReferenceWithExclusion>
+      <ProjectReference Remove="@(_transitiveProjectReferenceWithExclusion)"
+                        Condition="'%(_transitiveProjectReferenceWithExclusion.NuGetPackageId)' == '%(_transitiveProjectReferenceWithExclusion.Exclusion)'" />
     </ItemGroup>
   </Target>
 
-  <Target Name="RemoveConflictResolutionAssetsForTests"
-          AfterTargets="_HandlePackageFileConflicts"
-          DependsOnTargets="AddDefaultTestReferences"
-          Condition="'@(_defaultReferenceExclusionsFullPath)' != ''">
+  <!-- Make shared framework assemblies not app-local (non private). -->
+  <Target Name="UpdateProjectReferencesWithPrivateAttribute"
+          AfterTargets="AssignProjectConfiguration"
+          BeforeTargets="PrepareProjectReferences"
+          Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' and
+                     ('$(IsTestProject)' == 'true' or '$(IsTestSupportProject)' == 'true') and
```

Comment:
Is the `IsTest*` condition necessary?

Reply:
P2Ps are private by default for ref and src projects which makes this target a no-op for such.